pgmpy/estimators/ScoreCache.py

```diff
@@ -109,3 +122,51 @@ def __call__(self, *key):
             link[_PREV] = last
             link[_NEXT] = tail
         return value
+
+
+def check_scoring_method(scoring_method, data, use_cache):
```

Comment:
This function would fit the best in `StructureScore.py` where the classes K2, AIC etc are defined. But it is not possible, since it causes circular import. I put the function in `ScoreCache.py` to avoid the circular import. Alternatively, I could move the function in a new file.